runtime/major_gc.c

```diff
@@ -869,12 +1005,21 @@ void caml_init_major_heap (asize_t heap_size)
   caml_make_free_blocks ((value *) caml_heap_start,
                          Caml_state->stat_heap_wsz, 1, Caml_white);
   caml_gc_phase = Phase_idle;
-  gray_vals_size = 2048;
-  gray_vals = (value *) caml_stat_alloc_noexc (gray_vals_size * sizeof (value));
-  if (gray_vals == NULL)
-    caml_fatal_error ("not enough memory for the gray cache");
-  gray_vals_cur = gray_vals;
-  gray_vals_end = gray_vals + gray_vals_size;
+
+  Caml_state->mark_stack = caml_stat_alloc_noexc(sizeof(struct mark_stack));
+  if (Caml_state->mark_stack == NULL)
+    caml_fatal_error ("not enough memory for the mark stack");
+
+  Caml_state->mark_stack->stack =
+    caml_stat_alloc_noexc(MARK_STACK_INIT_SIZE * sizeof(mark_entry));
+
+  if(Caml_state->mark_stack->stack == NULL) {
+    caml_fatal_error("not enough memory for the mark stack");
+  }
```

Comment:
```suggestion
  if(Caml_state->mark_stack->stack == NULL)
    caml_fatal_error("not enough memory for the mark stack");
```

Cosmetic: coherence with above.